artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTProtocolManager.java

```diff
@@ -172,4 +177,8 @@ public void invokeIncoming(MqttMessage mqttMessage, MQTTConnection connection) {
    public void invokeOutgoing(MqttMessage mqttMessage, MQTTConnection connection) {
       super.invokeInterceptors(this.outgoingInterceptors, mqttMessage, connection);
    }
+
+   public Map<String, MQTTConnection> getConnectedClients() {
+      return connectedClients;
```

Comment:
I believe it would be good to return immutable view of the map.

Reply:
If I returned an immutable view of the map then I wouldn't be able to actually update it which would break org.apache.activemq.artemis.core.protocol.mqtt.MQTTConnectionManager#validateClientId.  I could add a new method (e.g. addConnectedClient(String, MQTTConnection)), but I don't see much benefit of doing that over what's happening now.

Reply:
I believe adding a method would be a good solution as it doesn't expose class internals.

Reply:
In my opinion it's not really exposing class internals.  The method simply returns a Map.  This same pattern is used all over the code-base.

Reply:
I agree that this pattern has been followed in all the places .If the map is only exposed to add entries to it, I feel it is always better to expose a method to do the same otherwise the map can get misused.